fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1247,30 +1354,46 @@ Reference<IDatabase> MultiVersionApi::createDatabase(const char *clusterFilePath
 		lock.leave();
 		throw network_not_setup();
 	}
-	lock.leave();
-
 	std::string clusterFile(clusterFilePath);
-	if(localClientDisabled) {
-		return Reference<IDatabase>(new MultiVersionDatabase(this, clusterFile, Reference<IDatabase>()));
+
+	if (threadCount > 1) {
```

Comment:
I think you also need to go in this path if there is 1 thread and local client is disabled, or at the very least you need to not create a local database.